packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -214,46 +217,47 @@ export const generateTypeDefGraphQLApi = async () => {
   }
 }
 
-export const generateTypeDefGraphQLWeb = async () => {
+function generateLoadTypedefsConfig(
+  generates: Record<
+    string,
+    CodegenTypes.ConfiguredOutput | CodegenTypes.ConfiguredPlugin[]
+  >
+) {
   const rwjsPaths = getPaths()
-  try {
-    const f = await runCodegenGraphQL({
-      [path.join(rwjsPaths.web.types, 'graphql.d.ts')]: {
-        documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
-        plugins: [
-          {
-            typescript: {
-              enumsAsTypes: true,
-            },
-          },
-          'typescript-operations',
-        ],
-      },
-    })
-    return f
-  } catch (e) {
-    console.error()
-    console.error('Error: Could not generate GraphQL type definitions (web)')
-    console.error()
-    return []
+
+  const ignore = []
+  for (const generatePath of Object.keys(generates)) {
+    if (path.extname(generatePath) === '') {
+      // we omit paths that don't resolve to a specific file
+      continue
+    }
+    ignore.push(path.join(process.cwd(), generatePath))
   }
-}
 
-/**
- * This is the function used internally by generateTypeDefGraphQLApi and generateTypeDefGraphQLWeb
- * And contains the base configuration for generating gql types with codegen
- *
- * Named a little differently to make it easier to spot
- */
+  const loaders = [
+    new CodeFileLoader({
+      pluckConfig: {
+        skipIndent: true,
+      },
+    }),
+  ]
 
-// CLI generate also takes cwd
-type CodegenConfig = CodegenTypes.Config & { cwd: string }
+  const loadTypedefsConfig: LoadTypedefsOptions<{ cwd: string }> = {
+    ignore,
+    loaders,
+    cwd: rwjsPaths.base,
+  }
 
-const runCodegenGraphQL = async (
-  generates: Record<string, CodegenTypes.ConfiguredOutput>
-) => {
+  return loadTypedefsConfig
+}
+
+function generateCodegenConfig(
```

Comment:
The naming of this function is misleading I think.

In this file "generate" refers to actually producing files in the user's project.

I would suggest renaming this to `getCodegenGeneratesConfig` or `getCodegenParams` or something similar

Reply:
Thanks Danny. Definitely need a better name here.

I looked at the function we're passing this config to and based the new name on that

```
export declare function generate(input: CodegenContext | (Types.Config & {
    cwd?: string;
}), saveToFile?: boolean): Promise<Types.FileOutput[] | any>;
```

So the new name I went with is `getCodegenInput`